WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionTable.js

```diff
@@ -0,0 +1,190 @@
+    && !sortByAnnouncement;
+}
+
+function RegistrationStatus({ comp }) {
+  if (comp.registration_status === 'not_yet_opened') {
+    return (
+      <Popup
+        trigger={<Icon className="clock blue" />}
+        content={I18n.t('competitions.index.tooltips.registration.opens_in', { duration: comp.timeUntilRegistration })}
+        position="top center"
+        size="tiny"
+      />
+    );
+  }
+  if (comp.registration_status === 'past') {
+    return (
+      <Popup
+        trigger={<Icon className="user times red" />}
+        content={I18n.t('competitions.index.tooltips.registration.closed', { days: I18n.t('common.days', { count: calculateDayDifference(comp.start_date, comp.end_date, 'future') }) })}
+        position="top center"
+        size="tiny"
+      />
+    );
+  }
+  if (comp.registration_status === 'full') {
+    return (
+      <Popup
+        trigger={<Icon className="user clock orange" />}
+        content={I18n.t('competitions.index.tooltips.registration.full')}
+        position="top center"
+        size="tiny"
+      />
+    );
+  }
+
+  return (
+    <Popup
+      trigger={<Icon className="user plus green" />}
+      content={I18n.t('competitions.index.tooltips.registration.open')}
+      position="top center"
+      size="tiny"
+    />
+  );
+}
+
+function DateIcon({ comp, showRegistrationStatus, sortByAnnouncement }) {
+  let tooltipInfo = '';
+  let iconClass = '';
+
+  if (comp.isProbablyOver) {
+    if (comp.resultsPosted) {
+      tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.posted');
+      iconClass = 'check circle result-posted-indicator';
+    } else {
+      tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.ended', { days: I18n.t('common.days', { count: calculateDayDifference(comp.start_date, comp.end_date, 'past') }) });
+      iconClass = 'hourglass end';
+    }
+  } else if (comp.inProgress) {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.in_progress');
+    iconClass = 'hourglass half';
+  } else if (sortByAnnouncement) {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.announced_on', { announcement_date: comp.announcedDate });
+    iconClass = 'hourglass start';
+  } else if (showRegistrationStatus) {
+    return <RegistrationStatus comp={comp} />;
+  } else {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.starts_in', { days: I18n.t('common.days', { count: calculateDayDifference(comp.start_date, comp.end_date, 'future') }) });
+    iconClass = 'hourglass start';
+  }
+
+  return (
+    <Popup
+      trigger={<Icon className={iconClass} />}
+      content={tooltipInfo}
+      position="top center"
+      size="tiny"
+    />
+  );
+}
+
+// Currently, the venue attribute of a competition object can be written as markdown,
+// and using third party libraries like react-markdown to parse it requires to much work
+function VenueMarkdown({ venueText }) {
```

Comment:
Function name should make it clear that it's not real markdown. And it's generic, not venue-related, so no need to mention 'venue' anywhere.
```suggestion
function PseudoMarkdown({ text }) {
```

Reply:
True, I will say this is still specific to only parsing link markdown `[]()`. It probably makes more sense to also move this to `lib/utils`.